streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -142,6 +146,25 @@ public StreamTask(final TaskId id,
     }
 
     /**
+     * Finds the source of the given topic partition.
+     *
+     * @param partition the topic partition
+     * @return
+     */
+    private SourceNode findSource(final TopicPartition partition) {
```

Comment:
Hmm, maybe it's better to fix these tests than modifying the production-code? E.g. we can add some test-only function to fill the topics from the pattern in `@before` rule.